File: jcrm-client-web/src/main/java/con/ss/jcrm/client/web/resource/ContactSiteResource.java
package con.ss.jcrm.client.web.resource;

import com.ss.jcrm.client.api.ContactSite;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.jetbrains.annotations.NotNull;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class ContactSiteResource {

    private String url;
    private String type;

    public ContactSiteResource(@NotNull ContactSite site) {
        this.url = site.getUrl();
        this.type = site.getType().name();
    }
}

File: ui-angular/src/app/resources/contact-site-resource.ts
import {ContactSite} from '@app/entity/contact-site';

export class ContactSiteResource {

    public static valueOf(site: ContactSite): ContactSiteResource {
        return new ContactSiteResource(site.url, site.type);
    }

    constructor(
        public url: string,
        public type: string
    ) {
    }
}
